Makefile.inc1

```diff
@@ -1904,25 +1904,22 @@ distributekernel distributekernel.debug: .PHONY
 	false
 .endif
 	mkdir -p ${DESTDIR}/${DISTDIR}
-.if defined(NO_ROOT)
-	@echo "#${MTREE_MAGIC}" > ${DESTDIR}/${DISTDIR}/kernel.premeta
-.endif
+	rm -f ${DESTDIR}/${DISTDIR}/kernel.premeta
 	${_+_}cd ${KRNLOBJDIR}/${INSTALLKERNEL}; \
 	    ${IMAKEENV} ${IMAKE_INSTALL:S/METALOG/kernel.premeta/} \
 	    ${IMAKE_MTREE} PATH=${TMPPATH:Q} ${MAKE} KERNEL=${INSTKERNNAME} \
 	    DISTBASE=/kernel DESTDIR=${INSTALL_DDIR}/kernel \
 	    METALOG=${METALOG:S/METALOG/kernel.premeta/} \
 	    ${.TARGET:S/distributekernel/install/}
 .if defined(NO_ROOT)
-	@sed -e 's|^./kernel|.|' ${DESTDIR}/${DISTDIR}/kernel.premeta > \
-	    ${DESTDIR}/${DISTDIR}/kernel.meta
+	echo "#${MTREE_MAGIC}" > ${DESTDIR}/${DISTDIR}/kernel.meta
+	sed -e 's|^./kernel|.|' ${DESTDIR}/${DISTDIR}/kernel.premeta | \
```

Comment:
does the unsortedness happen only with -j? I'd be a bit surprised by nondeterminism without -j.

Reply:
I'm not sure - I've never had the patience to run without `-j`.

One potential source of non-determinism is when file lists are generated rather than declared. For example, some of the items returned by `grep -R '=.*find' --include 'Makefile*' sys` could potentially be non-determinstic. But I can't say for sure.

Reply:
It looks like those are all in sys/contrib (and so not actually used in our build), but indeed anything that returns files in filesystem order could be an issue.

I expect `-j` will result in nondeterminism and so the sorting is necessary, mainly just wondering if this is the only source of nondeterminism here.